task/conftest-clair/0.1/conftest-clair.yaml

```diff
@@ -36,5 +36,32 @@ spec:
                    failures: (.failures // [])|map(.metadata.details.name) | unique
                  }' $(workspaces.conftest-ws.path)/clair-vulnerabilities.json || true)
         echo "${HACBS_TEST_OUTPUT:-${HACBS_ERROR_OUTPUT}}" | tee $(results.HACBS_TEST_OUTPUT.path)
-  workspaces:
-    - name: conftest-ws
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "appstudio, hacbs"
+  name: get-clair-scan
+spec:
+  params:
+    - name: image-digest
+      description: Image digest to scan
+    - name: image-url
+      description: Url to image
+  steps:
+    - name: get-vulnerabilities
+      image: quay.io/redhat-appstudio/clair-in-ci:latest
+      script: |
+        #!/usr/bin/env bash
+
+        imagewithouttag=$(echo '$(params.image-url)' | sed "s/\(.*\):.*/\1/" | tr -d '\n')
+        # strip new-line escape symbol from parameter and save it to variable
+        imageanddigest=$(echo $imagewithouttag@'$(params.image-digest)')
+
+        [ -f /workspace/registry-auth/.dockerconfigjson ] && REGISTRY_ARGS="/workspace/registry-auth/"
+
+        clair-action report --image-ref=$imageanddigest --db-path=/tmp/matcher.db --format=quay --docker-config-dir=$REGISTRY_ARGS > $(workspaces.clair-ws.path)/clair-result.json
```

Comment:
Since the workspace(**conftest-ws** and **clair-ws**) is not required anymore, we can drop the part mentioning it: 
`$(workspaces.clair-ws.path)/clair-result.json`
Its also important to mention that get-clair-scan step, must preceeds **conftest-clair**.
One and only namespace that will remain is **registry-auth** since it is needed to get the report form private repos.

Reply:
I recommend using tekton native fs for sharing the output of `get-clair-scan`. You can use /tekton/home folder for this purpose.